core/templates/dev/head/components/common-layout-directives/common-elements/promo-bar.directive.ts

```diff
@@ -57,23 +58,22 @@ angular.module('oppia').directive('promoBar', [
               return false;
             }
           };
-
-          PromoBarService.getPromoBarData().then(function(promoBarObject) {
-            ctrl.promoBarIsEnabled = promoBarObject.promoBarEnabled;
-            ctrl.promoBarMessage = promoBarObject.promoBarMessage;
-          });
-
-          // TODO(bhenning): Utilize cookies for tracking when a promo is
-          // dismissed. Cookies allow for a longer-lived memory of whether the
-          // promo is dismissed.
-          ctrl.promoIsVisible = !isPromoDismissed();
-
           ctrl.dismissPromo = function() {
             ctrl.promoIsVisible = false;
             setPromoDismissed(true);
           };
-        }
-      ]
+          ctrl.$onInit = function() {
+            PromoBarService.getPromoBarData().then(function(promoBarObject) {
+              ctrl.promoBarIsEnabled = promoBarObject.promoBarEnabled;
+              ctrl.promoBarMessage = promoBarObject.promoBarMessage;
+            });
+
+            // TODO(bhenning): Utilize cookies for tracking when a promo is
```

Comment:
Would it be possible to file this as an issue and link to that issue from here? Feel free to mention in that issue that this was originally reported by BenHenning as a TODO. (Otherwise we'll never get to it.)

Reply:
Done. #8300